patient-portal-backend/src/main/java/com/vaadin/demo/repositories/PatientRepository.java

```diff
@@ -2,6 +2,22 @@
 
 import com.vaadin.demo.entities.Patient;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
 
 public interface PatientRepository extends JpaRepository<Patient, Long> {
+    @Query("SELECT COUNT(p), '21-30' AS c FROM Patient p WHERE (p.birthDate >= '1987-01-01' AND p.birthDate <= '1997-01-01') UNION " +
```

Comment:
We can't use stored Postgres function calls with SQL written inside of the @Query annotation. I agree we can replace hard-coded the dates (I thought about this too, it was a tradeoff of pragmatism vs merit). If we do that, we'll have this JPA method having 12 parameters (the different start and end dates).